modules/test-utils/src/generate-layer-tests.js

```diff
@@ -98,9 +98,10 @@ export function generateLayerTests({
   testCases.forEach(testCase => {
     testCase.title = wrapTestCaseTitle(testCase.title);
     testCase.onBeforeUpdate = onBeforeUpdate;
-    testCases.onAfterUpdate = _onAfterUpdate;
   });
 
+  testCases[2].onAfterUpdate = _onAfterUpdate;
```

Comment:
I am not quite sure about this, I am wondering is it necessary to call onAfterUpdate when there is no data?

Reply:
There can be many test cases, you cannot only add the callback to one of them.
You can check if data is empty inside `onAfterUpdate`.